Potestas/Potestas/Processors/SerializeProcessor.cs

```diff
@@ -1,28 +1,92 @@
-﻿using System;
+﻿using Newtonsoft.Json;
+using Potestas.Exceptions.ProcessorExceptions;
+using System;
+using System.IO;
 
 namespace Potestas.Processors
 {
     /* TASK. Implement processor which serializes IEnergyObservation to the provided stream.
      * 1. Use serialization mechanism here. 
      * 2. Some IEnergyObservation could not be serializable.
      */
-    public class SerializeProcessor : IEnergyObservationProcessor
+    public class SerializeProcessor<T> : IStreamProcessor<T> where T : IEnergyObservation
     {
-        public string Description => throw new NotImplementedException();
+        private Stream _stream;
+
+        public string Description => "Serialize observations to the provided stream.";
+
+        /// <summary>
+        /// Create instance of SerializeProcessor.
+        /// Use this ctor when want to specify a stream later in the onNextMethod().
+        /// </summary>
+        public SerializeProcessor() { }
+
+        /// <summary>
+        /// Create instance of SerializeProcessor with the given stream.
+        /// </summary>
+        /// <param name="stream">he stream for serialization process.</param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when <paramref name="stream"/> is null.
+        /// </exception>
+        public SerializeProcessor(Stream stream)
+        {
+            _stream = stream ?? throw new ArgumentNullException($"The {nameof(stream)} can not be null.");
+        }
 
         public void OnCompleted()
         {
-            throw new NotImplementedException();
+            if (_stream != null)
+            {
+                _stream.Dispose(); // ask : should we dispose the stream here? 
```

Comment:
As far as I know, I an observable has completed and something attempts subscribe to it should be an exception or something. So it is probably ok to call `dispose` here

Reply:
ok. thanks.